### ISSUE TITLE
spack build error for Albany

### ISSUE THREAD

**Issue by ikalash** [sandialabs/Albany#848]
It looks like starting yesterday, the nightly spack Albany build stopped working.  Here is the error: 

```
==> albany: Executing phase: 'cmake'
==> albany: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

3 errors found in build log:
     1038        _main_func(__doc__)
     1039      File "/scratch/albany/nightlySpackBuild/spack-stage/ikalash/spack-stage-albany-develop-sprfcg4bl7u2oiqauqqpuazgi2ovxndv/
             spack-src/cmake/CreateExportAlbany", line 178, in _main_func
     1040        run (bin_dir,install_lib_dir,generator)
     1041      File "/scratch/albany/nightlySpackBuild/spack-stage/ikalash/spack-stage-albany-develop-sprfcg4bl7u2oiqauqqpuazgi2ovxndv/
             spack-src/cmake/CreateExportAlbany", line 149, in run
     1042        raise ValueError
     1043    ValueError
  >> 1044    make[2]: *** [export_albany.in] Error 1
     1045    make[2]: Leaving directory `/scratch/albany/nightlySpackBuild/spack-stage/ikalash/spack-stage-albany-develop-sprfcg4bl7u2o
             iqauqqpuazgi2ovxndv/spack-build-sprfcg4'
  >> 1046    make[1]: *** [src/CMakeFiles/create_export_albany.dir/all] Error 2
     1047    make[1]: *** Waiting for unfinished jobs....
     1048    [ 99%] Linking CXX executable dummy
```

This appears related to recent changes to how Albany details get exported - @mperego , @bartgol , can you please confirm?  What puzzles me is why this started yesterday and not earlier, since I think all the changes related to the exporting were made earlier.  Thoughts?

The spack build is important, as it is being used by Matt's collaborators running MALI and will continue to be used under FAntSSIE.

**Comment by mperego** [comment]
It is indeed weird that it started yesterday. Is this run daily? I know Xylar made changes to the Spack build recently. Is it possible that made other changes yesterday?

**Comment by ikalash** [comment]
Yes, it's run daily.  I checked my emails and the build was successful on Tuesday and broke Wednesday for the first time.  @xylar: have you made changes to the E3SM spack ( git clone git@github.com:E3SM-Project/spack.git)  recently that may cause this error?  I could check out the spack code from Tues and try building again as a sanity check.

**Comment by xylar** [comment]
@ikalash , Yes, I rebased onto the latest spack release. I needed to do that for an E3SM-Unified release. I pinged you and @mperego on Slack about it but I guess you don't really check that. I forgot.

**Comment by ikalash** [comment]
@xylar .  I am really bad at slack, sorry.  Email is the best way to reach me, or github issues.  Hmmm, I'm puzzled as to why that would cause this error.  Is spack now pulling an older version of Albany rather than VOTD?  I can check this.

**Comment by xylar** [comment]
@ikalash, I'm bad about email so GitHub it must be...

**Comment by ikalash** [comment]
So when I diff package.py for albany in spack VOTD vs. spack 18f66ff409eff777e2a77a4799e6d6ed243939d2, there are a lot of differences.  It seems the file got overwritten with what was in the main spack, maybe?  I would think you would keep the changes to the E3SM spack version for packages like Albany when you do the rebaselines/merges.

**Comment by xylar** [comment]
@ikalash, I certainly meant to. I was surprised how clean the rebase came out. I took out some changes to the main trilinos because they got reverted. That just makes the rebase easier. But I didn't deliberately alter Albany. But there were a handful of conflicts to resolve. I tried to always favor the changes over the upstream package. The best bet would be to look at one of the older spack_for_mache branches (maybe 1.4.0) since this would be before the rebase. We can copy back that package.py and commit it if you think that's easiest. Or cherry-pick any missing commits.

**Comment by xylar** [comment]
> here are a lot of differences. It seems the file got overwritten with what was in the main spack, maybe? I would think you would keep the changes to the E3SM spack version for packages like Albany when you do the rebaselines/merges.

I don't see any appreciable differences between `spack_for_mache_1.4.1`, which was essentially before the rebase and `develop`:
```
$ diff spack_for_mache_1.4.1/var/spack/repos/builtin/packages/albany/package.py develop/var/spack/repos/builtin/packages/albany/package.py
1c1
< # Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
---
> # Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
86c86
<         if '+sfad' in spec: 
---
>         if '+sfad' in spec:
```
Are you talking about a different package besides albany?

**Comment by xylar** [comment]
Also, the rebase happened on Sunday so it wouldn't explain differences between Tuesday and Wednesday.

**Comment by ikalash** [comment]
Shoot, I diff'ed the wrong files.  There are no diffs then.  Hmmm, this is very puzzling, especially given the timeline.

**Comment by ikalash** [comment]
@bartgol : can you please help me troubleshoot this error?  When I look more closely at the output, I see this: 

could not locate lib: /scratch/albany/nightlySpackBuild/spack-stage/ikalash/spack-stage-albany-develop-sprfcg4bl7u2oiqauqqpuazgi2ovxndv/spack-build-sprfcg4/dummy/-L/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/9.2.0/base/lib64
cwd: /scratch/albany/nightlySpackBuild/spack-stage/ikalash/spack-stage-albany-develop-sprfcg4bl7u2oiqauqqpuazgi2ovxndv/spack-build-sprfcg4/dummy


The first line doesn't make sense.  This is a path that exists /scratch/albany/nightlySpackBuild/spack-stage/ikalash/spack-stage-albany-develop-sprfcg4bl7u2oiqauqqpuazgi2ovxndv/spack-build-sprfcg4/dummy/, as is /projects/sems/install/rhel7-x86_64/sems/compiler/gcc/9.2.0/base/lib64.  Why is there a -L in between if the whole thing is supposed to be a path?  It seems that for some reason the path here is not being generated correctly, but I am not sure what you are trying to do with it exactly or what it should be, so it's hard for me to troubleshoot it.

**Comment by ikalash** [comment]
I have some more information:

(1) This is not due to changes to spack.  I verified I pulled an older version of spack, I get the same error.  No idea why it did not show up earlier... 
(2) I've verified that this issue is due to changes recently made by @bartgol relating to the exporting of Albany information.  If I checkout 7300b0ca5bd6e71cd4d61edd30b21a5417953a84 of Albany and cherry-pick some recent commits to avoid compilation errors, the code builds.  The exporting stuff was changes much earlier than this issue started, which is puzzling, but it is what it is.
(3) If I comment out line 149 of cmake/CreateExportAlbany, everything builds.  The issue is that library path not being populated with the right thing.

I am assigning to @bartgol since the changes that broke this were made by him.
